### PR TITLE
WRQ-14550: Add PageViews component

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
There's a need for a component to present screen to navigate page-by-page content with previous/next button.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add PageViews component

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Is the current implementation correct if the `pageIndicatorType` is `number` in the rtl region?
ltr region: 1/4 (number type page indicator)
rtl region: 4/1 (number type page indicator)

### Links
[//]: # (Related issues, references)
WRQ-14550

### CODE REVIEW

File: PageViews/tests/PageViews-specs.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			);
+
+			const nextButton = screen.getByLabelText('Next');
+
+			expect(nextButton).toBeInTheDocument();
+		}
+	);
+
+	test(
+		'should show previous button on the last page',
```
Comment (0x64) [inline PageViews/tests/PageViews-specs.js:25]: Let's test if the next button does not show also.

File: PageViews/PageViews.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+		delete rest.componentRef;
+		delete rest.onNextClick;
+		delete rest.onPrevClick;
+		delete rest.totalIndex;
+
+		return (
+			<div {...rest}>
+				{pageIndicatorType === 'dot' ? steps : null}
+				<Column style={{overflow: 'hidden'}}>
+					<Row style={{height: '100%'}}>
```
Comment (0x64) [inline PageViews/PageViews.js:242]: Let's use LESS file rather than in-line style.

File: PageViews/PageViews.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+							arranger={arranger}
+							component={ViewManager}
+							duration={400}
+							id="PageViewsContent"
+							index={index}
+							noAnimation={noAnimation}
+							onTransition={onTransition}
+							onWillTransition={onWillTransition}
+							reverseTransition={reverseTransition}
+							style={{overflow: 'hidden'}}
```
Comment (0x64) [inline PageViews/PageViews.js:254]: Let's use LESS file rather than in-line style.

File: PageViews/PageViews.js [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+			</div>
+		);
+	}
+});
+
+const PageViewsDecorator = compose(
+	Changeable({prop: 'index'}),
+	SpotlightContainerDecorator({
+		continue5WayHold: true,
+		defaultElement: [`.${spotlightDefaultClass}`, `#PageViewsContent *`, '#NextNavButton', '#PrevNavButton'],
```
Comment (0x64) [inline PageViews/PageViews.js:271]: Is it safe to using 'id' for searching navigable targets?

File: PageViews/PageViewsRouter.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
+import {useRef, useCallback, Children} from 'react';
+import EnactPropTypes from '@enact/core/internal/prop-types';
+import useChainRefs from '@enact/core/useChainRefs';
+import PropTypes from 'prop-types';
+
+import {useAutoFocus} from '../internal/Panels';
+
+import useFocusOnTransition from '../WizardPanels/useFocusOnTransition';
+import useToggleRole from '../WizardPanels/useToggleRole';
```
Comment (0x64) [inline PageViews/PageViewsRouter.js:9]: What about moving this hook out of WizardPanels? Its implementation is not dependent on any other component including WizardPanels.

File: PageViews/tests/PageViews-specs.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+		'should show next button on the first page',
+		() => {
+			render(
+				<PageViews index={0}>
+					<Page>I gots contents</Page>
+					<Page>I gots contents</Page>
+				</PageViews>
+			);
+
+			const nextButton = screen.getByLabelText('Next');
```
Comment (0x64) [inline PageViews/tests/PageViews-specs.js:18]: Let's test if the previous button is hidden on the first page.

File: PageViews/tests/PageViews-specs.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+		'should show previous button on the last page',
+		() => {
+			render(
+				<PageViews index={1}>
+					<Page>I gots contents</Page>
+					<Page>I gots contents</Page>
+				</PageViews>
+			);
+
+			const prevButton = screen.getByLabelText('Previous');
```
Comment (0x64) [inline PageViews/tests/PageViews-specs.js:34]: Let's test if the next button is hidden on the last page.

File: PageViews/Page.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * @class Page
+ * @memberof sandstone/PageViews
+ * @ui
+ * @public
+ */
+function Page ({
+	children
+}) {
+	return <div style={{height: '100%'}}>{children}</div>;
```
[the remark this answers is not in the document]
Comment (juwonjeong) [reply PageViews/Page.js]: I agree. So I tried to add css which can be applied to Page. Would it be fine?

File: PageViews/PageViews.js [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+
+		delete rest.componentRef;
+		delete rest.onNextClick;
+		delete rest.onPrevClick;
+		delete rest.totalIndex;
+
+		return (
+			<div {...rest}>
+				{pageIndicatorType === 'dot' ? steps : null}
+				<Column style={{overflow: 'hidden'}}>
```
[the remark this answers is not in the document]
Comment (juwonjeong) [reply PageViews/PageViews.js]: It's fixed

File: PageViews/PageViews.js (hunk 4 above)
Comment (juwonjeong) [reply PageViews/PageViews.js]: I fixed to use class for searching navigable targets.

File: PageViews/tests/PageViews-specs.js (hunk 6 above)
Comment (juwonjeong) [reply PageViews/tests/PageViews-specs.js:18]: It's added.

File: PageViews/Page.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (0x64) [reply PageViews/Page.js]: 👍🏼

**Review by 0x64 — APPROVED** [review]

**Pull request merged**